quinn-proto/src/endpoint.rs

```diff
@@ -861,6 +862,15 @@ pub struct TransportConfig {
     pub loss_reduction_factor: u16,
     /// Number of consecutive PTOs after which network is considered to be experiencing persistent congestion.
     pub persistent_congestion_threshold: u32,
+    /// Number of keep-alive packets to send before a connection times out.
+    ///
+    /// Keep-alive packets prevent an inactive but otherwise healthy connection from timing
+    /// out. They are distributed evenly over the idle timeout, and are only sent when the
+    /// connection is otherwise idle. Larger numbers will be more robust to packet loss.
+    ///
+    /// 0 to disable, which is the default. Only one side of any given connection should have
```

Comment:
Not sure I agree that `0` is a good default here -- doesn't TCP usually keep your connection alive for a while by default? Also, what happens in case both connections set this, this seems to imply some adverse effect? (Or is it saying that only one side needs to set it for the connections to stay alive?)

Reply:
I believe TCP's default behavior is no keep-alives and no idle timeout, i.e. connections only time out due to unacked data. I don't have a strong feeling about what defaults are correct; it strikes me as application-specific. Perhaps we should disable the idle timeout by default for consistency with TCP?

Reply:
Actually, that's not right either--disabling the idle timeout would be even more liberal than TCP, since QUIC does not terminate the connection due to excess retransmits.

Reply:
Discussion on slack suggests that issuing keepalives at intervals of <=20 seconds might be useful for preserving NAT bindings (see https://dl.acm.org/citation.cfm?id=1879174). That would mostly only be useful for preserving connections to servers which have connection IDs disabled, but those might not be too rare, if people are trying to minimize overhead.